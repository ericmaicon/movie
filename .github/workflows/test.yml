name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
