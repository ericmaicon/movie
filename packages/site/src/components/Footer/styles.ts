import styled from 'styled-components';

export const Footer = styled.div`
  background: url('/images/header.jpg') no-repeat center center fixed;
  background-size: cover;
  margin-top: 50px;
  min-height: 30px;
`;
