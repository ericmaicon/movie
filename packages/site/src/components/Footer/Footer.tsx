import React from 'react';

import * as Styled from './styles';

interface FooterProps {}

export default function Footer({ ...rest }: FooterProps) {
  return <Styled.Footer {...rest} />;
}
