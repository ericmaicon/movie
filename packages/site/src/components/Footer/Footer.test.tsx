import React from 'react';

import { render } from '~/test/index';

import Footer from './Footer';

describe('Footer', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(<Footer data-testid="footer" />);

    const footerElement = getByTestId('footer');

    expect(footerElement).toBeInTheDocument();
    expect(footerElement).toMatchSnapshot();
  });
});
