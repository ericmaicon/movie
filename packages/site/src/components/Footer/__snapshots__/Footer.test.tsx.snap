// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
.c0 {
  background: url('/images/header.jpg') no-repeat center center fixed;
  background-size: cover;
  margin-top: 50px;
  min-height: 30px;
}

<div
  class="c0"
  data-testid="footer"
/>
`;
